pkg/profiler/cpu/bpf_metrics.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package cpu
+
+import (
+	"bufio"
+	"bytes"
+	_ "embed"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strconv"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs"
+
+	bpf "github.com/aquasecurity/libbpfgo"
+)
+
+func GetBPFMetrics(logger log.Logger, m *bpf.Module, p procfs.Proc, mapMetrics *prometheus.GaugeVec) {
+	var bpfMapsNames []string
+	it := m.Iterator()
+
+	for {
+		mapBpf := it.NextMap()
+		if mapBpf != nil {
+			bpfMapsNames = append(bpfMapsNames, mapBpf.Name())
+		} else {
+			break
+		}
+	}
+
+	for _, mapName := range bpfMapsNames {
+		bpfMap, err := m.GetMap(mapName)
+		if err != nil {
+			level.Debug(logger).Log("msg", "error fetching bpf map", "err", err)
+			continue
+		}
+
+		bpfMaxEntry := fmt.Sprint(bpfMap.GetMaxEntries())
+		bpfMapKeySize := fmt.Sprint(bpfMap.KeySize())
+		bpfMapValueSize := fmt.Sprint(bpfMap.ValueSize())
+		bpfMapFd := fmt.Sprint(bpfMap.FileDescriptor())
+
+		bpfMemlock, err := FDInfoMemlock(logger, bpfMapFd, p)
+		if err != nil {
+			level.Debug(logger).Log("msg", "error getting memory locked for file descriptor", "err", err)
+		}
+
+		mapMetrics.WithLabelValues(mapName, bpfMapKeySize, bpfMapValueSize, bpfMaxEntry).Set(float64(bpfMemlock))
+	}
+}
+
+// FDInfoMemlock returns the memory locked by the fd for a process
+func FDInfoMemlock(logger log.Logger, fd string, p procfs.Proc) (int, error) {
+	path := fmt.Sprintf("/proc/%d/fdinfo/", p.PID) + fd
+	data, err := ReadFileNoStat(path)
+	if err != nil {
+		level.Debug(logger).Log("msg", "Unable to read fds for agent process", "agent_pid", p.PID, "err", err)
+	}
+
+	memlock := regexp.MustCompile(`^memlock:\s+(\d+)$`)
+
+	var text, memlockValue string
+
+	scanner := bufio.NewScanner(bytes.NewReader(data))
+	for scanner.Scan() {
+		text = scanner.Text()
+		if memlock.MatchString(text) {
+			memlockValue = memlock.FindStringSubmatch(text)[1]
+		}
+	}
+
+	memlockValueInt, err := strconv.Atoi(memlockValue)
+	if err != nil {
+		level.Debug(logger).Log("msg", "error converting memlock to integer type", "err", err)
+	}
+	return memlockValueInt, nil
+}
+
+// This is an intenal function from the prometheus/procfs/utils pkg
+func ReadFileNoStat(filename string) ([]byte, error) {
```

Comment:
```suggestion
func readFileNoStat(filename string) ([]byte, error) {
```

No need to export the method, it's only used here.